### PR TITLE
#282 - Hide Windows Menubar

### LINKED ISSUE #282: Hide menubar in Windows 10
Looks like this at the moment in Windows (possibly in linux as well)
butter

[here's](https://electron.atom.io/docs/api/browser-window/#winsetmenubarvisibilityvisible-windows-linux) some info that could help you fix this.

Great work guys, loving the app. This was literally the only "flaw" I've managed to find haha.

### PR DESCRIPTION
**Pull request by sallar**
Fixes #282

### CODE REVIEW

File: src/main/menu.js [hunk 1]
```diff
@@ -147,6 +147,10 @@ export function setupMenu(store) {
   const archives = getAllArchives(state);
   const currentArchiveId = getCurrentArchiveId(state);
   let condenced = Boolean(getSetting(state, 'condencedSidebar'));
+  let menubarVisibility = getSetting(state, 'menubarVisibility');
```
Comment (perry-mitchell) [inline src/main/menu.js]: Would naming this `menubarVisible` make more sense? Just because of how it's used..

File: src/main/windows.js [hunk 2]
```diff
@@ -24,6 +25,10 @@ export function setupWindows() {
       vibrancy: 'ultra-dark'
     });
 
+    // Set initial menu bar visibility
+    const menubarVisibility = getSetting(store.getState(), 'menubarVisibility');
+    win.setMenuBarVisibility(typeof menubarVisibility === 'boolean' ? menubarVisibility : true);
```
Comment (perry-mitchell) [inline src/main/windows.js]: Perhaps the naming here too..

File: src/shared/reducers/settings.js [hunk 3]
```diff
@@ -25,7 +25,8 @@ export function settingsByArchiveId(state = {}, action) {
 
 const DEFAULT_SETTINGS = {
   columnSizes: { tree: 230, entries: 230 },
-  condencedSidebar: true
+  condencedSidebar: true,
+  menubarVisibility: true
```
Comment (perry-mitchell) [inline src/shared/reducers/settings.js]: Yeah.. I think `menubarIsVisible` or `menubarVisible` might make more sense as it's boolean. What do you feel?

Comment (sallar) [reply src/shared/reducers/settings.js]: @perry-mitchell I named it after the method that sets it, but yeah we can change it. `menubarVisible` sounds good

**Comment by sallar** [comment]
@perry-mitchell done

**Pull request merged**